hyperledger/iroha
# [feature] #3276: Add `Log` isi

## Description

`Log` is was edded to make debugging of isi only triggers easier.





### Linked issue



Closes #3276  



### Benefits

Debugging should be easier.

### Drawbacks

We add new isi which will need to be maintained.

### How to test 

1. Submit `Log` isi to iroha through `cat log_isi.json | ./iroha_client_cli json`

```json
[
  {"Log": {"LogLevel":"ERROR","msg":{"String":"Cool error"}}}
]
```

2. Grep iroha message for this log message
3. Check that it have appropriate log level

### Migrate WASM to use `Log` isi instead of `log` function

After discussion with @mversic and @Arjentix it was decided not to make this migration because `Log` isi approach face multiple issues in `WASM` context: 

1. We lose `Span`s.
4. Not every WASM entrypoint is allowed to execute instructions at all.
5. Since every instruction is executed by validator we can't distinguish whatever this isi came from smart-contract or validator itself.  

Unless this problems will be solved migration is not desirable.



### Checklist

- [x] Move log `Level` into data model
- [x] Implement `Log` isi
